pkg/kubectl/namespace_test.go

```diff
@@ -59,3 +59,28 @@ func TestNamespaceGenerate(t *testing.T) {
 		}
 	}
 }
+
+func TestNamespaceGenerateNoName(t *testing.T) {
+	tests := []struct {
+		params    map[string]interface{}
+		expected  *api.Namespace
+		expectErr bool
+	}{
+		{
+			params: map[string]interface{}{
+				"name": 1,
+			},
+
+			expectErr: true,
+		},
+	}
+	generator := NamespaceGeneratorV1{}
+	for _, test := range tests {
```

Comment:
> i tried and the test cases was becoming little ugly, keeping it separate might be cleaner. Added some more negative use cases

Help me understand.  I would have expected this tweak in the loop above:

```go
switch{
case test.expectErr && err != nil:
    continue // loop, since there's no output to check
case test.expectErr && err == nil:
    t.Errorf("%v: expected error and didn't get one", index)
    continue // loop, no expected output object
case !test.expectErr && err != nil:
    t.Errorf("%v: expected error and didn't get one", index)
    continue // loop, no output object
case !test.expectErr && err == nil:
    // do nothing and drop through
}
```

I think that covers the four cases and keeps us all unified.